src/plugin.js

```diff
@@ -38,21 +38,27 @@ function NpmInstallPlugin(options) {
 NpmInstallPlugin.prototype.apply = function(compiler) {
   this.compiler = compiler;
 
+  var plugin = { name: "NpmInstallPlugin" };
+
   // Recursively install missing dependencies so primary build doesn't fail
```

Comment:
@insin Can you add a check here for webpack < 4 like this 
```js
if (compiler.hooks && compiler.hooks.watchRun )  { // all other hooks being used
 // webpack 4
}else{
// webpack 3
}
```
Thanks!
also Is this PR gonna merge ?!?

Reply:
I'm also wondering what is preventing this from being updated.  I see it was failing a test and without diving in it looks like the `"before each hook" should "checkBabel"` error could be fixed or worked around easily.  

At this point this is a `webpack-contrib` plugin referred to in official docs but is incompatible with the current version of Webpack.  If there is a reason the PR solution can't be adapted and merged, I think @insin should publish this in its own repo.

The process of trying to find a plugin that did this using `v4` was pretty confusing.  I searched npm and eventually found the forked version.  It's readme was the same as this repos, aside from mentioning that it was an unmerged PR for `v4` support at the top.  The repo link in that readme goes to this repo which is pretty much dead at this point.  

So, If this isn't going to be merged, why?  And if not, the [webpack-plugin-install-deps](https://www.npmjs.com/package/webpack-plugin-install-deps) should have it's own repo and updated readme indicating `4` support.  Either way, it still makes more since for this being a `webpack-contrib` repo instead of leaving this obsolete one up and not having a solution for `v4` and since @insin just wanted to contribute and indicated he would prefer not to maintain a separate repo, especially since this repo would just fade  into limbo without current support.